workshops/dodge/README.md

```diff
@@ -2,54 +2,696 @@
+  if (isGameOver) {
+    isGameOver = false;
+    player.position.x = width/2;
+    player.position.y = height-25;
+    enemy.position.x = width/2;
+    enemy.position.y = 0;
+  }
+}
+```
+
+Now we're in business. Mouse clicks will only reset the game if the game has ended, i.e., only on the game over screen.
+
+### Spicing it Up
+
+Let's make Dodge beautiful. We're going to bring the game from looking like this:
+
+![Before spicing it up](img/before_spicing_it_up.gif)
+
+To looking like this:
+
+![After we add images](img/after_spicing_it_up.gif)
+
+#### Adding Images
+
+To start, let's customize the game with images.
+
+We're providing the following images and have already uploaded them to an image host, giving us a URL to load the images from in our code.
+
+- _Player_
+
+  ![Player image](img/prophet_orpheus.png)
+
+  `https://surrogate.hackedu.us/i.imgur.com/N5uCbDu.png`
+
+- _Enemy_
+
+  ![Enemy image](img/asteroid.png)
+
+  `https://surrogate.hackedu.us/i.imgur.com/OdL0XPt.png`
+
+- _Background_
+
+  ![Background image](img/background.png)
+
+  `https://surrogate.hackedu.us/i.imgur.com/aKQOg3G.png`
+
+p5.js has a function called [`loadImage()`](http://p5js.org/reference/#p5/loadImage) that takes URL of an image as an argument and gives us a loaded image ready to be used. p5.play sprites have the method [`addImage()`](http://p5play.molleindustria.org/docs/classes/Sprite.html#method-addImage) that we can give a loaded image to assign it to the sprite.
+
+In addition to `setup()` and `draw()`, p5.play has a special function called `preload()`, which is run right when the page loads before `setup()`. `preload()` is generally used to load images and other resources (like sounds) into the game. Go ahead and declare `preload()` right before the `setup()` function.
+
+```js
+function preload() {
+}
+
+function setup() {
+  ...
+}
+```
+
+##### Customizing the Player
+
+Start by making a variable to hold our player image. Create this right below where we declared our `player` variable at the top of the file.
+
+```js
+...
+var player;
+var playerImage;
+...
+```
+
+Now let's load an image into `playerImage` in `preload()`.
+
+```js
+function preload() {
+  playerImage = loadImage("https://surrogate.hackedu.us/i.imgur.com/N5uCbDu.png");
+}
+```
+
+Add the loaded image to the `player`, right under where we create the player's sprite in `setup()`.
+
+```js
+function setup() {
+  ...
+  player = createSprite(width/2, height-25, 50, 50);
+  player.addImage(playerImage);
+  ...
+}
+```
+
+When we add an image to a sprite, the sprite's width and height is changed to match the image's. Because we're loading our sprite's image right after we create the sprite, we don't need to worry about the sprite's initial width and height. Let's just go ahead and set them to zero.
+
+```js
+function setup() {
+  ...
+  p̶l̶a̶y̶e̶r̶ ̶=̶ ̶c̶r̶e̶a̶t̶e̶S̶p̶r̶i̶t̶e̶(̶w̶i̶d̶t̶h̶/̶2̶,̶ ̶h̶e̶i̶g̶h̶t̶-̶2̶5̶,̶ ̶5̶0̶,̶ ̶5̶0̶)̶;̶
+  player = createSprite(width/2, height-25, 0, 0);
```

Comment:
Yeah, I figure that it's clearer to not introduce the concept of loosely
typed arguments yet.

On Fri, Apr 8, 2016 at 8:14 PM, JESSICA <notifications@github.com> wrote:

> In workshops/dodge/README.md
> <https://github.com/hackclub/hackclub/pull/781#discussion_r59109398>:
>
> > +```js
> > +function setup() {
> > +  ...
> > +  player = createSprite(width/2, height-25, 50, 50);
> > +  player.addImage(playerImage);
> > +  ...
> > +}
> > +```
> > +
> > +When we add an image to a sprite, the sprite's width and height is changed to match the image's. Because we're loading our sprite's image right after we create the sprite, we don't need to worry about the sprite's initial width and height. Let's just go ahead and set them to zero.
> > +
> > +```js
> > +function setup() {
> > +  ...
> > +  p̶l̶a̶y̶e̶r̶ ̶=̶ ̶c̶r̶e̶a̶t̶e̶S̶p̶r̶i̶t̶e̶(̶w̶i̶d̶t̶h̶/̶2̶,̶ ̶h̶e̶i̶g̶h̶t̶-̶2̶5̶,̶ ̶5̶0̶,̶ ̶5̶0̶)̶;̶
> > +  player = createSprite(width/2, height-25, 0, 0);
>
> I guess this is better than telling people that you don't have to include
> those arguments at all? Hah.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly or view it on GitHub
> <https://github.com/hackclub/hackclub/pull/781/files/bb7e11407098f5cf6ca44a9e996b10f9884834b0#r59109398>
>



-- 
Zach Latta
https://zachlatta.com <http://zachlatta.com>